File: sort/insert.py
def insertSort(a):
    for i in range(len(a) - 1):
        j = i - 1
        temp = a[i]
        while (j > -1 and a[j] > temp):
            a[j + 1] = a[j]
            j -= 1
        a[j + 1] = temp
    return a


#冒泡排序测试
nums = [32, 31, 29, 26, 20, 24, 35, 6, 25, 39]
print("sort before:")
print(nums)
nums = insertSort(nums)
print("sort after:")
print(nums)

File: sort/insert.cpp
#include <iostream>
using namespace std;

void insertSort(int a[], int n) {
    for(int i = 0; i < n; i++) {
        int j = i - 1;
        int temp = a[i];
        while(j > -1 && a[j] > temp) {
            a[j + 1] = a[j];
            j --;
        }
        a[j + 1] = temp;
    }
}

void output(int a[], int n)
{
    for (int i = 0; i < 10; i++)
    {
        cout << a[i] << " ";
    }
    cout << endl;
}

int main() {
    int a[] = {32, 31, 29, 26, 20, 24, 35, 6, 25, 39};
    cout << "sort before:" << endl;
    output(a, 10);
    insertSort(a, 10);
    cout << "sort after:" << endl;
    output(a, 10);
    return 0;
}